### PR TITLE
Add strict CIDR check in IP Pool validator

### PR DESCRIPTION
**Pull request by xychu**
## Description
PR for #425 

## Todos
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: lib/validator/validator.go [hunk 1]
```diff
@@ -47,6 +47,7 @@ var (
 	reasonString        = "Reason: "
 	poolSmallIPv4       = "IP pool size is too small (min /26) for use with Calico IPAM"
 	poolSmallIPv6       = "IP pool size is too small (min /122) for use with Calico IPAM"
+	poolUnStictCIDR     = "IP pool CIDR is not strictly masked"
```
Comment (robbrockbank) [inline lib/validator/validator.go]: UnStict -> Unstrict

File: lib/validator/validator.go [hunk 2]
```diff
@@ -348,6 +349,16 @@ func validateIPPool(v *validator.Validate, structLevel *validator.StructLevel) {
 			}
 		}
 
+		// The Calico CIDR should be strictly masked
+		if !pool.Spec.Disabled {
```
Comment (robbrockbank) [inline lib/validator/validator.go]: I don't think we should check the Disabled flag here - we should prevent configuration or disabled pools too.

**Comment by xychu** [comment]
@robbrockbank  thanks for your review and I've updated the code.
Also add one test for this.

**Comment by robbrockbank** [comment]
LGTM - than you @xychu for the contribution.

**Pull request merged**